### PR TITLE
Create tournament

### PR DESCRIPTION
**Pull request by larsvandriel**
Adds User and Tournament functionality to the api.

The added functionality contains a Create and Read function to User and Tournament.

### CODE REVIEW

**Comment by MindHawk** [comment]
Which issues does this adress (if not close, which ones does it touch upon?)

**Review by Zeehond99 — DISMISSED** [review]

File: src/main/java/com/fontys/api/entities/Tournament.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{
+    @GeneratedValue @Id
+    private Integer id;
+    private String name;
+    private String description;
+    @OneToOne
+    private User user;
+    private int numberOfTeams;
+
+    public Tournament() {}
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/entities/Tournament.java]: would be unnecessary when using @NoArgsConstructor

**Review by lmtjanssen — DISMISSED** [review]

File: src/main/java/com/fontys/api/entities/Tournament.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public Tournament(String name, String description, User user, int numberOfTeams)
+    {
+        this.name = name;
+        this.description = description;
+        this.user = user;
+        this.numberOfTeams = numberOfTeams;
+    }
+
+    public Tournament(Integer id, String name, String description, User user, int numberOfTeams)
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/entities/Tournament.java]: would be unnecessary when using @AllArgsConstructor

File: src/main/java/com/fontys/api/entities/User.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import javax.persistence.Id;
+
+@Data
+@Entity
+public class User
+{
+    @GeneratedValue @Id
+    private Integer id;
+
+    public User() {}
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/entities/User.java]: would be unnecessary when using @NoArgsConstructor

File: src/main/java/com/fontys/api/queries/TournamentQuery.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    private final TournamentService tournamentService;
+
+    public TournamentQuery(TournamentService tournamentService)
+    {
+        this.tournamentService = tournamentService;
+    }
+
+    public List<Tournament> tournaments() { return tournamentService.tournaments(); }
+
+    public Optional<Tournament> tournament(Integer id) { return tournamentService.tournament(id); }
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/queries/TournamentQuery.java:24]: Search can only be performed on id?

File: src/main/java/com/fontys/api/queries/UserQuery.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@Component
+public class UserQuery implements GraphQLQueryResolver
+{
+    private final UserService userService;
+
+    public UserQuery(UserService userService) { this.userService = userService; }
+
+    public List<User> users(int count) { return this.userService.getAllUsers(); }
+
+    public Optional<User> user(Integer id) { return this.userService.getUser(id); }
```
Comment (lfb0801) [inline src/main/java/com/fontys/api/queries/UserQuery.java:21]: Search can only be performed on id?

File: src/main/resources/graphql/tournament.graphqls [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    id: ID!,
+    name: String,
+    description: String,
+    owner: User,
+    numberOfTeams: Int,
+}
+
+type Query {
+    tournaments(count: Int):[Tournament]
+    tournament(id: ID):Tournament
```
Comment (lfb0801) [inline src/main/resources/graphql/tournament.graphqls]: shouldn't ID have ! marking?

File: src/main/resources/graphql/user.graphqls [hunk 7]
```diff
@@ -0,0 +1,12 @@
+type User {
+    id: ID!,
+}
+
+type Query {
+    users(count: Int):[User]
+    user(id: ID):User
```
Comment (lfb0801) [inline src/main/resources/graphql/user.graphqls]: shouldn't ID have ! marking?

**Review by lfb0801 — CHANGES_REQUESTED** [review]
A lot of work means a lot of potential for improvement

**Review by MindHawk — DISMISSED** [review]
Implement @lfb0801 's changes OR defend your code

**Review by MichielLangen — CHANGES_REQUESTED** [review]
I agree with the changes requested by @lfb0801

File: src/main/java/com/fontys/api/queries/UserQuery.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Optional;
+
+@Component
+public class UserQuery implements GraphQLQueryResolver
+{
+    private final UserService userService;
+
+    public UserQuery(UserService userService) { this.userService = userService; }
+
+    public List<User> users(int count) { return this.userService.getAllUsers(); }
```
Comment (chf3kl9) [inline src/main/java/com/fontys/api/queries/UserQuery.java]: We're not using pagination, so 'int count' is not needed

**Review by chf3kl9 — CHANGES_REQUESTED** [review]

**Review by lfb0801 — DISMISSED** [review]

**Review by MichielLangen — DISMISSED** [review]
***nice***

File: src/main/java/com/fontys/api/queries/UserQuery.java (hunk 8 above)
Comment (larsvandriel) [reply src/main/java/com/fontys/api/queries/UserQuery.java]: Fixed the problem. Is it  better now?

**Review by lfb0801 — APPROVED** [review]

**Review by MichielLangen — APPROVED** [review]

**Review by chf3kl9 — APPROVED** [review]

**Review by Zeehond99 — APPROVED** [review]

**Review by Zeehond99 — APPROVED** [review]

File: src/test/java/com/fontys/api/mockrepositories/MockJPARepository.java [hunk 9]
```diff
@@ -0,0 +1,489 @@
+package com.fontys.api.mockrepositories;
```
Comment (MindHawk) [inline src/test/java/com/fontys/api/mockrepositories/MockJPARepository.java:1]: Bit big?

**Review by MindHawk — APPROVED** [review]

File: src/test/java/com/fontys/api/mockrepositories/MockJPARepository.java (hunk 9 above)
Comment (larsvandriel) [reply src/test/java/com/fontys/api/mockrepositories/MockJPARepository.java:1]: It's as small as it's going to be. It's mostly code that is from the JPARepository itself.

**Review by lmtjanssen — APPROVED** [review]

**Pull request merged**